MINOR: Only update a request's local complete time in API handler if unset.

bdbyrne opened the pull request:
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

Commit:
MINOR: Only update a request's local complete time in API handler if unset.

hachikuji commented on core/src/main/scala/kafka/server/KafkaApis.scala:
```diff
@@ -176,7 +176,8 @@ class KafkaApis(val requestChannel: RequestChannel,
     } finally {
-      request.apiLocalCompleteTimeNanos = time.nanoseconds
+      if (request.apiLocalCompleteTimeNanos < 0)
+        request.apiLocalCompleteTimeNanos = time.nanoseconds
```
This actually seems like an intuitive time to set the local complete time. I guess the reason we set it in `RequestChannel.Response` also is that we are trying to set it consistently with `responseCompleteTimeNanos`. I can't really think of a better solution than what you have here, but it might at least be helpful to add a comment of explanation?

ijuma replied:
I was looking at this code and the way it works today is super confusing. :) I think this change makes sense. However, I was curious how you noticed this. Was there a delay in getting to this point somehow? In theory, the value at this point should be very similar to the value we set during processing.

bdbyrne replied:
From what I remember, the value was close enough that I didn't consider it worth looking into. I encountered it when I added debug metrics to further subdivide the local time, and there appeared to be lost time, i.e. the sum wasn't adding to the whole.

ijuma replied:
Makes sense, thanks.

Commit:
Add comment.

bdbyrne commented:
It's also updated from the ClientRequestQuotaManager when determining the throttle time. I added a general comment stating the somewhat-obvious that it's possible for it to be set during processing, but in the end, I don't think it's a big deal. I only noticed it when doing some profiling and finding some numbers were off in the calculations.

hachikuji reviewed (approved):
Thanks, LGTM

hachikuji commented:
retest this please
